openquake/calculators/scenario_risk.py

```diff
@@ -34,36 +33,29 @@ def scenario_risk(riskinputs, param, monitor):
     """
     Core function for a scenario computation.
 
-    :param riskinput:
-        a of :class:`openquake.risklib.riskinput.RiskInput` object
+    :param riskinputs:
+        a list of :class:`openquake.risklib.riskinput.RiskInput` objects
     :param param:
         dictionary of extra parameters
     :param monitor:
         :class:`openquake.baselib.performance.Monitor` instance
     :returns:
         a dictionary {
-        'agg': array of shape (E, L, R, 2),
-        'avg': list of tuples (lt_idx, rlz_idx, asset_ordinal, statistics)
-        }
-        where E is the number of simulated events, L the number of loss types,
-        R the number of realizations  and statistics is an array of shape
-        (n, R, 4), with n the number of assets in the current riskinput object
+        'alt': AggLoggTable instance
+        'avg': list of tuples (lt_idx, rlz_idx, asset_ordinal, statistics)}
     """
     crmodel = monitor.read('crmodel')
     result = dict(avg=[], alt=param['alt'])
     for ri in riskinputs:
         for out in ri.gen_outputs(crmodel, monitor, param['tempname']):
-            num_events = param['num_events'][out.rlzi]
             param['alt'].aggregate(
                 out, param['minimum_asset_loss'], param['aggregate_by'])
             for l, loss_type in enumerate(crmodel.loss_types):
                 losses = out[loss_type]
-                if numpy.product(losses.shape) == 0:  # happens for all NaNs
-                    continue
                 avg = numpy.zeros(len(ri.assets), F64)
                 for a, asset in enumerate(ri.assets):
                     aid = asset['ordinal']
-                    avg[a] = av = losses[a].sum() / num_events
+                    avg[a] = av = losses[a].sum()
```

Comment:
We should probably change the name of this variable, since it is no longer storing the average itself? This naming may be confusing for future readers of the code who won't know there's a multiplier coming up ahead.

Reply:
Indeed. I will have to think of something.